oneview_redfish_toolkit/client_session.py

```diff
@@ -44,6 +49,49 @@ def init_map_clients():
     globals()['map_clients'] = OrderedDict()
 
 
+def init_gc_for_expired_sessions():
+    gc_thread = threading.Thread(target=_gc_for_expired_sessions)
+    gc_thread.daemon = True
+    gc_thread.start()
+
+
+def _gc_for_expired_sessions():
+    time.sleep(GC_FREQUENCY_IN_SEC)
+    logging.info('Verifying expired sessions...')
+
+    tokens_to_remove = []
+    map_items = _get_map_clients().items()
+    for token, dict_by_token in map_items:
+        ov_clients = iter(dict_by_token['client_ov_by_ip'].values())
+        ov_client = next(ov_clients)  # ov_clients can be in any order
+        ov_session_id = ov_client.connection.get_session_id()
+        try:
+            # request that does not increment the life of Oneview session
+            resp, body = ov_client.connection.do_http(
+                'GET',
+                '/rest/sessions/',
+                '',
+                {'Session-Id': ov_session_id}
+            )
+
+            # when is not a success
+            if resp.status not in range(200, 300):
+                if resp.status == 404:
+                    tokens_to_remove.append(token)
+                else:
+                    logging.error('Unexpected response with status {} '
+                                  'of Oneview sessions endpoint: {}'
+                                  .format(resp.status, body))
+
+        except Exception as e:
+            logging.exception('Unexpected error: {}'.format(e))
+
+    for token in tokens_to_remove:
+        clear_session_by_token(token)
+
+    _gc_for_expired_sessions()
```

Comment:
I think that it's better use and while loop, because if the same method is called repeatedly in an recursive way without never finish the previous execution, the program execution will have to allocate memory for the local variables for each recursive call. Then since the old method execution doesn't finish and instead of that it's calling a new method execution, the allocated memory for the local variable from the previous method execution can not be released, so it can cause a memory leakage.

Reply:
yeaah! OMG, I forgot the problem with the stack of call. :sweat:
Thanks. =)